storage/innobase/row/row0sel.cc

```diff
@@ -3318,8 +3318,9 @@ row_sel_store_mysql_rec(
 		row_mysql_prebuilt_free_blob_heap(prebuilt);
 	}
 
-	if (UNIV_LIKELY_NULL(prebuilt->compress_heap))
-		row_mysql_prebuilt_free_compress_heap(prebuilt);
+	if (UNIV_LIKELY_NULL(prebuilt->compress_heap)) {
```

Comment:
The original way is using empty. 
We can use `row_mysql_prebuilt_free_compress_heap`, it will work as well, sure. But if we are looping over rows, it will cause freeing all blocks and allocating again. If we empty instead of free, it is a good chance that already allocated 1st block is big enough, so we will avoid allocation.

Reply:
I see, an optimization.  

Can you please execute the test case from PS-7657? with and without optimization?

Reply:
I did it locally. Both versions work fine.